src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -67,12 +67,20 @@ public static int LeadingZeroCount(uint value)
                 return ArmBase.LeadingZeroCount(value);
             }
 
-            // Unguarded fallback contract is 0->31
+            // Unguarded fallback contract is 0->31, BSR contract is 0->undefined
             if (value == 0)
             {
                 return 32;
             }
 
+            if (X86Base.IsSupported)
+            {
+                // LZCNT returns index starting from MSB, whereas BSR gives the index from LSB.
+                // The XOR here is equivalent to subtraction since the BSR result is always between 0 and 31.
+                // This saves one instruction, as subtraction from constant requires either MOV/SUB or NEG/ADD.
+                return 31 ^ (int)X86Base.BitScanReverse(value);
```

Comment:
Should we track this as a more general JIT optimization?

Reply:
It only applies in very limited situations.  LHS has to be a power of 2 minus one and RHS has to be between 0 and LHS.  We know those are both true here, but I don't think the JIT could know that on its own anywhere.